zetaclient/evm/evm_signer.go

```diff
@@ -324,246 +332,233 @@ func (signer *Signer) SignCommandTx(
+		logger.Warn().Msgf("gasLimit %d is too high; set to %d", cctx.GetCurrentOutTxParam().OutboundTxGasLimit, txData.gasLimit)
 	}
 
-	logger.Info().Msgf("chain %s minting %d to %s, nonce %d, finalized zeta bn %d", toChain, cctx.InboundTxParams.Amount, to.Hex(), nonce, cctx.InboundTxParams.InboundTxFinalizedZetaHeight)
-	sendHash, err := hex.DecodeString(cctx.Index[2:]) // remove the leading 0x
-	if err != nil || len(sendHash) != 32 {
-		logger.Error().Err(err).Msgf("decode CCTX %s error", cctx.Index)
-		return
-	}
-	var sendhash [32]byte
-	copy(sendhash[:32], sendHash[:32])
-
-	// use dynamic gas price for ethereum chains
-	var gasprice *big.Int
-
+	// use dynamic gas price for ethereum chains.
 	// The code below is a fix for https://github.com/zeta-chain/node/issues/1085
 	// doesn't close directly the issue because we should determine if we want to keep using SuggestGasPrice if no OutboundTxGasPrice
 	// we should possibly remove it completely and return an error if no OutboundTxGasPrice is provided because it means no fee is processed on ZetaChain
 	specified, ok := new(big.Int).SetString(cctx.GetCurrentOutTxParam().OutboundTxGasPrice, 10)
 	if !ok {
-		if common.IsEthereumChain(toChain.ChainId) {
-			suggested, err := signer.client.SuggestGasPrice(context.Background())
+		if common.IsEthereumChain(chain.ChainId) {
+			suggested, err := client.SuggestGasPrice(context.Background())
 			if err != nil {
-				logger.Error().Err(err).Msgf("cannot get gas price from chain %s ", toChain)
-				return
+				return errors.Join(err, fmt.Errorf("cannot get gas price from chain %s ", chain))
 			}
-			gasprice = roundUpToNearestGwei(suggested)
+			txData.gasPrice = roundUpToNearestGwei(suggested)
 		} else {
-			logger.Error().Err(err).Msgf("cannot convert gas price  %s ", cctx.GetCurrentOutTxParam().OutboundTxGasPrice)
-			return
+			return fmt.Errorf("cannot convert gas price  %s ", cctx.GetCurrentOutTxParam().OutboundTxGasPrice)
 		}
 	} else {
-		gasprice = specified
+		txData.gasPrice = specified
 	}
+	return nil
+}
+
+func SetTransactionData(
+	cctx *types.CrossChainTx,
+	evmClient *ChainClient,
+	evmRPC interfaces.EVMRPCClient,
+	logger zerolog.Logger,
+	zetaBridge interfaces.ZetaCoreBridger,
+	txData *TransactionData) (bool, error) {
+
+	txData.outboundParams = cctx.GetCurrentOutTxParam()
+	txData.amount = cctx.GetCurrentOutTxParam().Amount.BigInt()
+	txData.nonce = cctx.GetCurrentOutTxParam().OutboundTxTssNonce
+	txData.sender = ethcommon.HexToAddress(cctx.InboundTxParams.Sender)
+	txData.srcChainID = big.NewInt(cctx.InboundTxParams.SenderChainId)
+	txData.asset = ethcommon.HexToAddress(cctx.InboundTxParams.Asset)
+
+	skipTx := SetChainAndSender(cctx, logger, txData)
+	if skipTx {
+		return true, nil
+	}
+
+	toChain := common.GetChainFromChainID(txData.toChainID.Int64())
+	if toChain == nil {
+		return true, fmt.Errorf("unknown chain: %d", txData.toChainID.Int64())
+	}
+
+	// Get nonce, Early return if the cctx is already processed
```

Comment:
not sure why this would happen, but the check was here before. If the cctx was processed already it shouldn't be in the pending list but there may be some timing issues where the cctx wasn't updated but the tx was confirmed. I'm going to leave it for now.